pkg/start/start.go

```diff
@@ -195,96 +192,103 @@ func (o *Options) run(ctx context.Context, controllerCtx *Context, lock *resourc
 	controllerCtx.OpenshiftConfigManagedInformerFactory.Start(informersDone)
 	controllerCtx.InformerFactory.Start(informersDone)
 
-	resultChannelCount++
-	go func() {
-		defer utilruntime.HandleCrash()
-		leaderelection.RunOrDie(postMainContext, leaderelection.LeaderElectionConfig{
-			Lock:            lock,
-			ReleaseOnCancel: true,
-			LeaseDuration:   o.leaderElection.LeaseDuration.Duration,
-			RenewDeadline:   o.leaderElection.RenewDeadline.Duration,
-			RetryPeriod:     o.leaderElection.RetryPeriod.Duration,
-			Callbacks: leaderelection.LeaderCallbacks{
-				OnStartedLeading: func(_ context.Context) { // no need for this passed-through postMainContext, because goroutines we launch inside will use runContext
-					launchedMain = true
-					if o.ListenAddr != "" {
+	if !cache.WaitForCacheSync(informersDone, controllerCtx.CVO.GetCacheSynced()...) {
```

Comment:
nit: Instead of exposing `GetCacheSynced`, maybe the public API should be `controllerCtx.CVO.WaitForCacheSync(postMainContext)`?  Or... where are we're creating these informers?  Maybe we don't need to claw anything back out of the `CVO` property at all?